pyriemann_qiskit/utils/preprocessing.py

```diff
@@ -0,0 +1,69 @@
+from sklearn.base import TransformerMixin
+from sklearn.preprocessing import RobustScaler
+
+
+class NdRobustScaler(TransformerMixin):
+    """Apply one robust scaler by feature.
+
+    See [1]_ for more details.
+
+    References
+    ----------
+    .. [1] \
+        https://stackoverflow.com/questions/50125844/how-to-standard-scale-a-3d-matrix
+
+    Notes
+    -----
+    .. versionadded:: 0.2.0
+    """
+
+    def __init__(self):
+        self._scalers = []
+
+    """Fits one robust scaler on each feature of the training data.
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_matrices, n_features, n_samples)
+        Training matrices.
+    _y : ndarray, shape (n_samples,)
+        Unused. Kept for scikit-learn compatibility.
+
+    Returns
+    -------
+    self : NdRobustScaler instance
+        The NdRobustScaler instance.
+    """
+
+    def fit(self, X, _y=None, **kwargs):
+        _, n_features, _ = X.shape
+        self._scalers = []
+        for i in range(n_features):
+            scaler = RobustScaler().fit(X[:, i, :])
+            self._scalers.append(scaler)
+        return self
+
+    """Apply the previously trained robust scalers (on scaler by feature)
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_matrices, n_samples, n_features)
```

Comment:
X must have the same shape as in `fit()` docstring.

Reply:
In scikit-learn, RobustScaler uses `(n_samples, n_features)` inputs.
We should prefer `(n_matrices, n_samples, n_features)`, rather than `(n_matrices, n_features, n_samples)`.
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.RobustScaler.html

Reply:
Yes, but pyRiemann transformers are n_matrices x n_channels x n_times.
Same for the method to plot waveforms.

Reply:
Ok for me. But we must update the docstring
```suggestion
    X : ndarray, shape (n_matrices, n_features, n_samples)
```